pkg/utils/injection/injection.go

```diff
@@ -63,3 +63,19 @@ func GetConfig(ctx context.Context) *rest.Config {
 	}
 	return retval.(*rest.Config)
 }
+
+type componentNameKeyType struct{}
+
+var componentNameKey = componentNameKeyType{}
+
+func WithComponentName(ctx context.Context, name string) context.Context {
```

Comment:
Thoughts on 
```suggestion
func WithControllerName(ctx context.Context, name string) context.Context {
```

Reply:
The term "controller" was too course, e.g. the *provisioning* controller creates `Provisioner` objects which also create a `Packer` object, all use the `CloudProvider` instance.  Applying the label "controller" to calls attributed to `Provisioner` and `Packer` objects would be misleading, but it did  still seem preferable to attribute the calls to the responsible objects rather than rolling them all up to the *provisioning* controller.  So the more general term "component" was used.

Reply:
I think we *do* want to associate the calls with the controller, not the component. The individual implementations of packer/scheduler, etc, are non-fundamental to the code base. It's simply how the code happens to have been implemented and will (and has) change frequently. The controllers however, are fundamental in that they are essential to how to the system is architected. 

If we did simply want to know the caller, we could use `runtime.Caller(0)`. Of course, I think we should be controller-aware, instead.